### PR TITLE
fixes #20497 - host provisioning build mode enabled

### PR DESCRIPTION
**Pull request by timogoebel**

### CODE REVIEW

**Comment by lzap** [comment]
Tests finally fixed [test].

File: test/functional/discovered_hosts_controller_test.rb [hunk 1]
```diff
@@ -73,6 +73,13 @@ def test_edit_form_attributes
     assert_not_nil host.cpu_count
   end
 
+  def test_edit_form_build_mode_enabled
+    host = Host::Discovered.import_host(@facts)
```
Comment (lzap) [inline test/functional/discovered_hosts_controller_test.rb]: Oh one more thing, can you use this helper instead now:

    host = discover_host_from_facts(@facts)

**Review by lzap — CHANGES_REQUESTED** [review]
One test nitpick, thanks.

**Comment by timogoebel** [comment]
@lzap : Rebased and changed test to use the new helper.

**Comment by lzap** [comment]
Failures are weird, running locally.

**Comment by lzap** [comment]
I don't see these errors locally. Fantastic.

**Comment by lzap** [comment]
I really don't know why core tests fail randomly. @ares could this be related to the new permission validator?

I created a core patch to show Rails logs, maybe we will see something interesting there:

https://github.com/theforeman/foreman/pull/4739

**Comment by ares** [comment]
sorry, from expected false to be truthy I can't really tell... the permissions changes cause deterministic issues though

**Comment by lzap** [comment]
Random failures trying again with newer foreman develop [test]

**Comment by timogoebel** [comment]
once again - [test]

**Comment by lzap** [comment]
Can you please rebase once again or re-trigger? Develop now seems more stable, still random JS failures but not 600 failures....

**Comment by timogoebel** [comment]
[test]

**Comment by timogoebel** [comment]
@lzap: Is this ready to go? Tests are green.

**Comment by lzap** [comment]
Sure, works great thanks.

**Pull request merged**